lib/extras/liquid_view.rb

```diff
@@ -22,30 +22,27 @@ def initialize(view)
   def render(template, local_assigns = nil)
     @view.controller.headers["Content-Type"] ||= 'text/html; charset=utf-8'
 
-    # Rails 2.2 Template has source, but not locals
-    if template.respond_to?(:source) && !template.respond_to?(:locals)
-      assigns = (@view.instance_variables - PROTECTED_INSTANCE_VARIABLES).inject({}) do |hash, ivar|
-                  hash[ivar[1..-1]] = @view.instance_variable_get(ivar)
-                  hash
-                end
-    else
-      assigns = @view.assigns.reject{ |k,v| PROTECTED_ASSIGNS.include?(k) }
-    end
-
-    source = template.respond_to?(:source) ? template.source : template
-    local_assigns = (template.respond_to?(:locals) ? template.locals : local_assigns) || {}
+    assigns = @view.assigns
 
-    if content_for_layout = @view.instance_variable_get("@content_for_layout")
-      assigns['content_for_layout'] = content_for_layout
+    if @view.content_for?(:layout)
+      assigns["content_for_layout"] = @view.content_for(:layout)
     end
     assigns.merge!(local_assigns.stringify_keys)
 
-    liquid = Liquid::Template.parse(source)
-    liquid.render(assigns, :filters => [@view.controller.master_helper_module], :registers => {:action_view => @view, :controller => @view.controller})
+    controller = @view.controller
+    filters = if controller.respond_to?(:liquid_filters, true)
+                controller.send(:liquid_filters)
+              elsif controller.respond_to?(:master_helper_module)
+                [controller.master_helper_module]
+              else
+                [controller._helpers]
+              end
+
+    liquid = Liquid::Template.parse(template)
+    liquid.render(assigns, :filters => filters, :registers => {:action_view => @view, :controller => @view.controller})
```

Comment:
from https://stackoverflow.com/questions/7855837/how-to-make-liquid-file-work-in-rails-3